### PR TITLE
UX: Large image placeholder style changed in cooked html

### PR DESCRIPTION
**Pull request by vinothkannans**
Large image placeholder UI changed as inline in cooked html. **And totally removed the large images inside onebox**.

### CODE REVIEW

File: app/jobs/regular/pull_hotlinked_images.rb [hunk 1]
```diff
@@ -46,7 +48,8 @@ def execute(args)
       raw = post.raw.dup
       start_raw = raw.dup
       downloaded_urls = {}
-      broken_images, large_images = [], []
+      large_images = eval(post.custom_fields[LARGE_IMAGES].presence || "[]")
```
Comment (ZogStriP) [inline app/jobs/regular/pull_hotlinked_images.rb]: Why is `eval` needed here?

File: app/jobs/regular/pull_hotlinked_images.rb [hunk 2]
```diff
@@ -104,15 +108,18 @@ def execute(args)
 
       end
 
+      post.custom_fields[LARGE_IMAGES] = large_images.to_s
```
Comment (ZogStriP) [inline app/jobs/regular/pull_hotlinked_images.rb]: Why are you doing a `.to_s`? CustomField should support arrays. If they don't, it's best to serialize using JSON.

File: app/jobs/regular/pull_hotlinked_images.rb (hunk 1 above)
Comment (SamSaffron) [reply app/jobs/regular/pull_hotlinked_images.rb]: yes please ... absolutely no eval here.

File: lib/cooked_post_processor.rb [hunk 3]
```diff
@@ -81,11 +81,70 @@ def post_process_images
...
 
     images.each do |img|
+      next if large_images.include?(img["src"]) && add_large_image_placeholder!(img)
+
       limit_size!(img)
       convert_to_link!(img)
     end
   end
 
+  def add_large_image_placeholder!(img)
```
Comment (ZogStriP) [inline lib/cooked_post_processor.rb]: Nokogiri API is way too verbose. Is there a way we could inject HTML instead?

File: lib/cooked_post_processor.rb [hunk 4]
```diff
@@ -81,11 +81,70 @@ def post_process_images
...
+        lspan.add_previous_sibling(i)
+      end
+    end
+
+    img.remove
+    true
+  end
+
+  def large_images
+    @large_images ||= eval(@post.custom_fields[Jobs::PullHotlinkedImages::LARGE_IMAGES].presence || "[]")
```
Comment (ZogStriP) [inline lib/cooked_post_processor.rb]: Same here. It's best to avoid `eval`.

File: lib/cooked_post_processor.rb (hunk 3 above)
Comment (vinothkannans) [reply lib/cooked_post_processor.rb]: Yes. But we already using it in same class to create lightbox and I thought it's more clean comparing to raw HTML. I guess you feel like those repeated lines of Nokogiri are not necessary here. So I refactored everything by common methods. Also I changed some existing code using those methods. Please have a look now.

**Comment by SamSaffron** [comment]
@tgxworld is this good to merge?

**Comment by SamSaffron** [comment]
@ZogStriP is this good to merge now?

**Pull request merged**